cmd/osm-bootstrap/crds/policy_upstream_traffic_setting.yaml

```diff
@@ -162,6 +162,242 @@ spec:
+                              - host
+                              - port
+                              properties:
+                                host:
+                                  description: Hostname of the global rate limit service.
+                                  type: string
+                                  minLength: 1
+                                port:
+                                  description: Port of the global rate limit service.
+                                  type: integer
+                                  minimum: 1
+                                  maximum: 65535
+                            domain:
+                              description: Domain defines a container for a set of rate limits.
+                                All domains known to the Ratelimit service must be globally unique.
+                                They serve as a way to have different rate limit configurations that
+                                don't conflict.
+                              type: string
+                            descriptors:
+                              description: List of rate limit descriptors to use in the rate limit service request.
+                              type: array
+                              minItems: 1
+                              items:
+                                description: Rate limit descriptor to use in the rate limit service request.
+                                type: object
+                                required:
+                                - entries
+                                properties:
+                                  entries:
+                                    description: List of rate limit descriptor entries for the descriptor.
+                                    type: array
+                                    minItems: 1
+                                    items:
+                                      description: Descriptor entry.
+                                      type: object
+                                      required:
+                                      - key
+                                      - value
+                                      properties:
+                                        key:
+                                          description: Key of the descriptor entry.
+                                          type: string
+                                        value:
+                                          description: Value of the descriptor entry.
+                                          type: string
+                            timeout:
+                              description: Timeout (optional) interval for calls to the rate limit service. Defaults to 20ms.
+                              type: string
+                            failOpen:
+                              description: FailOpen (optional) defines whether to allow traffic in case of communication failure between
```

Comment:
For my own understanding could you explain why the default failOpen value is true?

Reply:
That's the default behavior in Envoy. So we tend to use Envoy defaults for optional values (within the controller) unless there is a reason not to.

Reply:
Got it. I see that Envoy's [`failure_mode_deny` is set to false ](https://www.envoyproxy.io/docs/envoy/v1.23.0/api-v3/extensions/filters/network/thrift_proxy/filters/ratelimit/v3/rate_limit.proto#extensions-filters-network-thrift-proxy-filters-ratelimit-v3-ratelimit) which makes sense. I agree with the default `failOpen` value being true. I was looking around at other implementations and I saw that the default `failOpen` value for [contour's rate limiting service api](https://projectcontour.io/docs/main/config/api/#projectcontour.io/v1alpha1.RateLimitServiceConfig) is false which is interesting. Thanks!